### PR TITLE
Fixes sync calling async subflows

### PR DESCRIPTION
**Pull request by ahuang11**
### Summary

Closes https://github.com/PrefectHQ/prefect/issues/6497

The async subflow gets handled downstream:
```
    # Sync subflow run
    if not parent_flow_run_context.flow.isasync:
        return run_async_from_worker_thread(begin_run)
    else:
        return parent_flow_run_context.sync_portal.call(begin_run)
```

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- No tests or issue needed
- [x] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [ ] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Review by bunchesofdonald — APPROVED** [review]
I was just looking at this issue and this is essentially what I was about to do, so it's a +1 from me.

**Comment by ahuang11** [comment]
> I was just looking at this issue and this is essentially what I was about to do, so it's a +1 from me.

Yes, right after I made a PR and was linking the issue, I saw that! Sorry I didn't self-assign it earlier, and thanks for helping review

File: src/prefect/engine.py
Comment (madkinsz) [inline src/prefect/engine.py:139]: Can you update the comments to mirror those for `enter_task_run_engine`?

**Review by madkinsz — COMMENTED** [review]
We'll need product sign-off on this. I don't remember why we weren't supporting this in the first place. cc @cicdw / @billpalombi

**Comment by billpalombi** [comment]
Good catch! I'll defer to @cicdw here but I don't think it was a deliberate decision not to support calling async flows from sync flows. As long as there aren't any unintuitive side effects, I totally support this.

**Review by cicdw — APPROVED** [review]
LGTM, agree with everyone here that this was most likely not intentional

**Pull request merged**
